packages/@aws-cdk/aws-ec2/lib/volume.ts

```diff
@@ -89,6 +89,17 @@ export interface EbsDeviceOptions extends EbsDeviceOptionsBase {
    * @default false
    */
   readonly encrypted?: boolean;
+
+  /**
+   * The ARN of the AWS Key Management Service (AWS KMS) CMK used for encryption.
+   *
+   * You have to ensure that the KMS CMK has the correct permissions to be used by the service launching the ec2 instances.
+   *
+   * @see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSEncryption.html#ebs-encryption-requirements
+   *
+   * @default - If encrypted is true, the default aws/ebs KMS key will be used.
+   */
+  readonly kmsKeyId?: string;
```

Comment:
> I don't like to only specify a kmsKeyId (= ARN) but accepting an kms.IKey in the properties would be a bigger change.

That appears to be true (the code in `ebs-util.ts` will need to be enhanced). However, it's the right thing to do. Accepting the IKey means that we can do things (in the future) like automatically granting the correct permissions to the instance/template. It also matches our API standards. If we accept the change like this now, moving to accepting an IKey will be a huge pain (for both us and our customers) versus just doing the right thing now.

Reply:
I thought about it and changed the implementation, so that it would be possible to handle transformations. What do you think?